opengrok-indexer/src/main/java/org/opengrok/indexer/util/TandemFilename.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opengrok.indexer.util;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+
+/**
+ * Represents a utility class for creating a filename to operate in tandem with
+ * an original filename by adding a new file extension but limiting the length
+ * of the new filename to 255 UTF-8 encoded bytes if necessary by truncating
```

Comment:
why is it necessary to leave the part of the filename in ? (as opposed to using just the digest)

Reply:
I prefer to be able to correlate files in `xref/` using just my human brain, so I chose to stick with a straight-forward concatenation for filenames that encode as anything up to 254 bytes and for anything longer to pack the digest into the tail of the name and limit the total to 255 bytes.

Reply:
Initially I thought you do not trust SHA-256 to produce unique digests :-) It certainly helps in tracking down which file is which.